### PR TITLE
Do not update apps if it comes from git

### LINKED ISSUE #7011: Dev work inside apps is overwritten by marketplace app version
As @Henni wrote in https://github.com/nextcloud/contacts/issues/305#issuecomment-339470033

> I lost my work when my dev instance overwrote my git repository during an update... 😢 

This shouldn’t happen.

@ChristophWurst suggested:

> It's simple: if an app is downgraded, then don't replace it. Alternatively, don't run the update everything that can be updated after an app update/downgrade.

I added:

> or is it possible to check if the app is checked out via git? In that case it should of course not be upgraded with one from the app store.

### PR DESCRIPTION
**Pull request by skjnldsv**
We had a lot of  issues as devs by those auto update...

Fixes #7011

### CODE REVIEW

File: lib/private/Installer.php [hunk 1]
```diff
@@ -399,6 +399,10 @@ public function isUpdateAvailable($appId) {
 			$apps = $this->appFetcher->get();
 		}
 
+		if ($this->isInstalledFromGit($appId) === true) {
+			return false;
+		}
```
Comment (MorrisJobke) [inline lib/private/Installer.php]: Could we move this check one up ... just to return earlier in the method?

Comment (skjnldsv) [reply lib/private/Installer.php]: Of course!

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Review by rullzer — APPROVED** [review]
Works for the 99% use case for me! Lets do this.

**Comment by ChristophWurst** [comment]
What happens if a developer accidentally packages the `.git` dir? Does that mean the app will never be updated again? 🙈

**Comment by skjnldsv** [comment]
@ChristophWurst Yup!
We should add a test in the appstore for that I think! :)

**Comment by MorrisJobke** [comment]
> We should add a test in the appstore for that I think! :)

nextcloud/appstore#537

**Comment by skjnldsv** [comment]
We could also refuse to install an app if it has a .git in it ;)

**Comment by MorrisJobke** [comment]
> What happens if a developer accidentally packages the .git dir? Does that mean the app will never be updated again? 🙈

@ChristophWurst @skjnldsv Another idea: only reject this if the nextcloud itself is also a git clone (update channel === git) 😉

**Comment by MorrisJobke** [comment]
@BernhardPosselt Mentioned that we maybe should also add checks for other typical VCS systems and have a blacklist:

* SVN
* HG
* bazaar
* git

Anything else?

**Comment by skjnldsv** [comment]
> @ChristophWurst @skjnldsv Another idea: only reject this if the nextcloud itself is also a git clone (update channel === git) 😉

@MorrisJobke I would say no, because I sometimes uses default apps without using their git repo :)

**Comment by jancborchardt** [comment]
cc @Henni for review as he also had this issue :)

**Comment by MorrisJobke** [comment]
Okay - let's go for the `.git` folder for now and check if this is already enough.

**Pull request merged**